tools/resourcedocsgen/cmd/docs/registry.go

```diff
@@ -215,6 +215,7 @@ func resourceDocsFromRegistryCmd() *cobra.Command {
 	}
 
 	cmd.Flags().StringVar(&baseDocsOutDir, "baseDocsOutDir", "../../content/registry/packages", "The directory path to where the docs will be written to")
+	cmd.Flags().StringVarP(&branch, "branch", "b", "master", "The branch at which the registry repo will be checked out.")
```

Comment:
Do we need this? I ask because a bit ago, after running `hugo mod get` to use your `praneetloke/regen-metadata` branch, I ran `make build`, and saw this in the logs:

```
I1112 11:31:14.667116   82648 registry.go:155] Cloning the registry repo @ master
I1112 11:31:17.552228   82648 registry.go:162] Cloned the repo successfully!
I1112 11:31:17.552255   82648 registry.go:165] Resetting registry repo HEAD to b91845d16a01
HEAD is now at b91845d1 Merge branch 'master' into praneetloke/regen-metadata
```

So it looks like branch isn't being used, so may not be necessary. What do you think? Do we need it?

Reply:
Yeah the flag itself isn't used by `build-site.sh`. The second message you are seeing in from `git` itself as a result of running `git reset --hard <commit>`. We don't need the `branch` flag in that it is strictly not needed for our purpose but I can see it being helpful for development purposes where someone wants to specify the branch from which to regen docs rather than a specific commit. If you specify both flags, then it becomes "checkout registry repo at that branch _and_ also reset the branch to that commit" but again we don't use that in CLI. However, it could be useful for development purposes.